### DISCUSSION TITLE
linux/clash

### DISCUSSION THREAD

**Discussion by giscus** [LaoYutang/blogGiscus#10] [Announcements]
# linux/clash

官方Clash部署需要采用两个镜像分别启动服务和面板，博主使用官方server镜像和官方管理面板前端代码，重新修改打包构建，使用一个镜像可以直接启动服务和管理面板，简单轻量。

https://blog.laoyutang.cn/linux/clash.html

**Comment by DongLiStudio** [comment]
有没有办法设置登录密码，没密码的话吧端口打开的话岂不是都可以改我clash的配置？还是说每次用的时候再把端口打开？

**Reply by LaoYutang** [reply]
因为是内网使用的服务所以没考虑登录密码的问题，可以考虑用nginx转发一下加上登录。另外提醒一下不要在公网搭建，国内属于违法行为。

**Reply by DongLiStudio** [reply]
好的，感谢解答

**Comment by zhoujiaming1996** [comment]
现在是不是不可用了

**Reply by zhoujiaming1996** [reply]
编辑外部控制设置，这里是什么意思

**Reply by LaoYutang** [reply]
可以用的，这玩意是Clash内核加上官方的web端集成出来的。编辑外部控制设置，这个没懂你意思

**Comment by SilverFox008** [comment]
接口请求为什么都是404

**Comment by sijw123** [comment]
编辑外部控制设置 该怎么配置

**Comment by iockx** [comment]
```sh
docker: Error response from daemon: driver failed programming external connectivity on endpoint clash (548b92fab9ba2440be780997d0c0cc33c312d94e05c6cb5232c98fee129f5bcc):  (iptables failed: iptables --wait -t nat -A DOCKER -p tcp -d 0/0 --dport 7888 -j DNAT --to-destination 172.17.0.2:8080 ! -i docker0: iptables: No chain/target/match by that name.
```
run之后 这是？

**Comment by molodye** [comment]
按照您的教程，输入ip：7888 打不开

**Reply by wkitt4** [reply]
我也遇到了，我是没有yaml文件，容器反复重启

**Reply by LaoYutang** [reply]
看下docker的日志报了什么错误把 ``` docker logs clash```

**Comment by molodye** [comment]
哥们 可以交流一下吗？ 我是portainer来设置的，我发现portainer只能对文件夹映射，没发对文件映射。这个问题怎么解决呀
&nbsp;


molodye
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年12月7日(星期四) 晚上8:06
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [LaoYutang/blogGiscus] linux/clash (Discussion #10)





 
我也遇到了，我是没有yaml文件，容器反复重启

**Reply by LaoYutang** [reply]
portainer应该可以映射文件的啊,   如果你只能映射文件夹, 那只能把/root/.config/clash整个文件夹映射出来了.  使用volume来映射, 然后把你的配置文件覆盖到volume里面去.
像这样 -v clash-config:/root/.config/clash

**Reply by AtticusZeller** [reply]
直接命令行运行就没问题

**Comment by stan0105xx** [comment]
我按照教程启动了 404是为什么?

**Reply by molodye** [reply]
我也是这样

**Reply by LaoYutang** [reply]
修复了nginx不启动问题

**Comment by fhtim** [comment]
这个镜像之支持amd64架构，不支持arm64的，所以很多人都没法正常启动。作者埋的坑，没说清楚。

**Reply by LaoYutang** [reply]
镜像已添加ARM支持

**Comment by molodye** [comment]
哦哦&nbsp; 应该是这样，真是醍醐灌顶。我用的是nanopc-T4&nbsp; cpu是arm64的rk3399&nbsp;


molodye
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年1月24日(星期三) 晚上9:17
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [LaoYutang/blogGiscus] linux/clash (Discussion #10)





 
这个镜像之支持amd64架构，不支持arm64的，所以很多人都没法正常启动。作者埋的坑，没说清楚。

**Reply by LaoYutang** [reply]
镜像已添加ARM支持

**Comment by jeizai** [comment]
大佬 用的是什么内核啊

**Reply by LaoYutang** [reply]
我是amd64的机器， 现在镜像arm64跟 armv7都加上了已经

**Comment by nice2cu1** [comment]
出现‘编辑外部控制设置’的原因是因为在创建容器之前没有把/data/clash/clash.yaml配置好，我这里因为直接运行创建容器的命令会直接创建一个名为clash.yaml的文件夹，然后docker把这个文件夹映射过去了，虽然服务起了，但是没有配置文件，进面板就会一直弹出“编辑外部控制设置”的对话框，得先把clash得配置文件放好再创建容器，否则日志会输出Can't find config, create a initial config file，当看到日志输出你的proxy groups名后就代表成功了。

另外，我想问一下要怎么配才让本机和其他容器走这个clash容器的代理呢

**Comment by AtticusZeller** [comment]
感谢大佬，已成功，Ubuntu22

**Reply by AtticusZeller** [reply]
但是记得在下面的

**remember set proxy** in `/etc/environment`
```bash
nano /etc/environment
```
-  add the following
```text
http_proxy="http://127.0.0.1:7890"
https_proxy="http://127.0.0.1:7890"
ftp_proxy="ftp://127.0.0.1:7890"
no_proxy="localhost,127.0.0.1,::1"
```
- apply changes
```bash
source /etc/environment
```
- test connection
```bash
curl https://www.google.com

**Reply by AtticusZeller** [reply]
我的命令是，调整了yaml配置命令的映射
```bash
docker run -d \
  --name clash \
  --restart=always \
  --log-opt max-size=1m \
  -v /root/.config/clash/config.yaml:/root/.config/clash/config.yaml \
  -p 7888:8080 -p 7890:7890 \
  laoyutang/clash-and-dashboard:latest
```

**Comment by molodye** [comment]
好的&nbsp; 谢谢啦。这个配置命令先收藏我再试试
&nbsp;


Joe
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Atticus ***@***.***&gt;; 
发送时间: 2024年2月3日(星期六) 下午2:12
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [LaoYutang/blogGiscus] linux/clash (Discussion #10)





 
我的命令是，调整了yaml配置命令的映射
 docker run -d \   --name clash \   --restart=always \   --log-opt max-size=1m \   -v /root/.config/clash/config.yaml:/root/.config/clash/config.yaml \   -p 7888:8080 -p 7890:7890 \   laoyutang/clash-and-dashboard:latest
 
&mdash;

**Comment by refrain1128** [comment]
yaml文件配置完，打开网页代理下面全是灰的，没法测速，yaml我照着电脑端v2ray里的数据填的，VLESS和Trojan各填了一个，没法测速也没法连接

**Comment by youzhi1233** [comment]
一直弹出《编辑外部控制设置》窗口

**Comment by leap233** [comment]
请问一下 我的容器启动成功了,使用代理也没问题,就是前端的管理页面打不开
这是我的启动命令
sudo docker run -itd \
  --name clash \
  --restart=always \
  --log-opt max-size=1m \
  -v $(pwd)/config_fny.yaml:/root/.config/clash/config.yaml \
  -p 5891:8080 -p 5890:7890 \
  laoyutang/clash-and-dashboard:latest


访问 htttp://xxxx:5891  显示无法访问此网站
但是5890的代理是可以使用的,
请问如何解决。

**Comment by ouooo** [comment]
大哥我真的服了，太感谢了，弄了几天只有你这个能成，我快哭了，收下我的膝盖

**Comment by xxx854** [comment]
Parse config error: external-ui: /root/.config/clash/dashboard not exist"

**Comment by iockx** [comment]
请问下，我的是乌班图22.04，我是用的是windows的clash，在乌班图上的浏览器可以使用外网，但是部署的程序却无法使用外网（程序是指定了7890端口的，而且本地运行正常，所以可以排除编写程序的问题）

难道浏览器和ssh访问外网有区别么？我是第一次遇到这个问题，大佬分析下？感谢

**Reply by LaoYutang** [reply]
描述似乎有点乱，我理解成你有一台 windows 电脑和一台 ubuntu 的服务器在同一局域网内。window上启动了 clash 服务，http 代理端口是 7890，程序在 windows 上运行是正常的，在 ubuntu 上调用代理失败了。
这种情况首先考虑代码里面是不是更好 http 代理的 ip 地址了，是用 curl -x 也可以排除代理在 ubuntu 上是否可用

**Reply by iockx** [reply]
对的，代码中访问外网的request都会指定代理；
今晚又研究了下，貌似找到了问题，我的clash如果启动的话，再启动docker容器，7890端口会冲突（docker中映射了7890端口），在想解决办法；

**Comment by Pigeva** [comment]
每次重启设置都重置了。能加上环境变量吗？大佬

比如局域网开关、代理端口这种

**Reply by Pigeva** [reply]
因为我的yaml文件配置   不是我需要的，每次都要手动改yaml或者设置  都很麻烦

**Reply by Pigeva** [reply]
因为我的yaml文件，里面包含的设置相关的配置 不是我需要的，所以每次都要手动改yaml或者设置 都很麻烦

**Reply by LaoYutang** [reply]
clash都是基于yaml文件配置的，你应该是机场自动下载的配置文件每次更新都需要改下？可以考虑写个脚本自动获取自动改配置，然后重启服务

**Reply by LaoYutang** [reply]
比如这个样子
```bash
#!/bin/bash
curl -o /data/clash/clash.yaml https://www.cokecloud.net/#/register?code=O60IwzOS
sed -i -E -e 's/^#?\s*port.*/port:\ 7890/' \
          -e 's/^#?\s*allow-lan.*/allow-lan:\ true/' /data/clash/clash.yaml
docker restart clash
```

**Comment by howrain** [comment]
dashboard提示“编辑外部控制设置”是clash.yaml没有配置external-controller，配置默认的9090端口就行

**Comment by SuJinchuan** [comment]
能不能新增个多配置源管理及定时同步管理的模块

**Comment by 851032763** [comment]
[31mFATA[0m[0030] Initial configuration directory error: can't initial MMDB: can't download MMDB: Get "https://cdn.jsdelivr.net/gh/Dreamacro/maxmind-geoip@release/Country.mmdb": dial tcp 46.82.174.69:443: i/o timeout 
请问大佬，报这个无法下载可咋办

**Comment by javelin266** [comment]
在7888页面选择节点和开放端口后，并不会在本地开始socks的端口和http的端口，只有7890的端口是映射出去的

**Comment by bennywan** [comment]
容器部署顺利，yaml文件中导入的所有代理地址也能看到也能测速，但是最上面“开机时启动”和“设置为系统代理”两个开关不让点击是怎么回事？鼠标放上去的时候是红色禁止的图标。

**Reply by bennywan** [reply]
容器跑起来以后，在宿主机执行 export http_proxy=http://127.0.0.1:7890 \ export https_proxy=http://127.0.0.1:7890 ，也不行，用不了代理

**Reply by bennywan** [reply]
解决了，容器是没有权限控制宿主机的。只能在宿主机上用命令行设置。上面的export必须放在脚本文件里用source指令执行。

**Comment by xhh1238** [comment]
本地无法访问管理页面勾选允许局域网连接 怎么办  没有图形化

**Comment by pingban404** [comment]
这个网址挂了https://cloudupup05.com/auth/register?code=g9Rq，我想看看配置模板来着😂

**Comment by cleanerbaba** [comment]
好用，谢谢大佬

**Comment by cleanerbaba** [comment]
评论提醒自己“勾选允许局域网连接”🥲

**Comment by zjboy1984** [comment]
大佬，我装完，打开管理页面，显示要我填9090的那个外部控制，但我点确定没有任何反应呢

**Comment by defaultuserJ** [comment]
大佬yyds！！！

**Comment by Jianghengyf** [comment]
设置-设置为系统代理 这个按钮点不开是怎么回事呀

**Reply by Jianghengyf** [reply]
是不是因为这按钮点不开，所以我电脑设置了代理地址也不能科学上网？

**Reply by LaoYutang** [reply]
点不了按钮是因为设置中的系统代理是指给docker容器的系统代理。电脑在系统代理里设置http代理应该是没问题的，检查下机场状态和软件走不走系统代理吧

**Comment by wodty** [comment]
网页已经登入，请注意，修改该配置项并不会修改你的 Clash 配置文件，请确认修改后的外部控制地址和 Clash 配置文件内的地址一致，否则会导致 Dashboard 无法连接。

这个地方怎么配置？填写添加 之后点不了确定

**Reply by xycqrcg0** [reply]
可能是clash配置文件里allow-lan设为false了，改为true应该就行了

**Comment by th8664484** [comment]
我部署好以后使用ip:端口可以访问成功。改成域名访问，nginx代理就提示要修改配置，external-controller 这个改成域名也不行，是什么问题，感谢
